### PR TITLE
Error boundaries

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

@mxstbr would love your feedback on this. Basically I want to make sure that issues like #3127 don't crash the entire app - obviously we have a lot of components that have the possibility of throwing errors, and if those bubble up to the top level error boundary in `routes.js` it means the whole app is in an error state.

So I went through and wrapped a bunch of components in error boundaries. The granularity here is up to us, but my philosophy here was: what are the things that we can live with silently failing in order to preserve the core experience? E.g. if a single message crashes, we error that message not the whole thread. If you're viewing a community and the channel list crashes, we don't error the whole community view. A large result of this mindset is failing individual items in lists - e.g. threads, channels, users, etc. so that if one item in a list fails we don't take the whole list down with it.

Anyways - open to feedback here both on implementation of the error boundary and the granularity of implementation. Hopefully this will help provide a bit more stability around core experiences (reading threads, writing messages) even if tertiary components fail for some reason.

### CODE REVIEW

File: src/components/error/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  render() {
+    const { error } = this.state;
+    const { fallbackComponent: FallbackComponent, children } = this.props;
+
+    if (error) {
+      if (this.props.fallbackComponent) {
+        // $FlowFixMe
+        return <FallbackComponent />;
```
Comment (mxstbr) [inline src/components/error/ErrorBoundary.js:30]: We have a component like this already, no need to create your own: `import ErrorBoundary from 'react-error-boundary'`

Literally does the exact same thing 😅 Docs: https://github.com/bvaughn/react-error-boundary

**Review by mxstbr — DISMISSED** [review]
Overall this makes sense to me—I trust your judgement on the granularity of these. Let's just use the component we already have instead of our own and ship this!

**Comment by brianlovin** [comment]
I'm ~90% certain that the error boundary is swallowing errors and not reporting them to sentry. So actually we should replace that npm module with our own coded one which handles the sentry calls.

**Comment by brianlovin** [comment]
CI Failure is memory failure...:(

**Pull request merged**